tests/available_solvers.cpp

```diff
@@ -149,38 +125,20 @@ SmtSolver create_interpolating_solver(SolverEnum se)
 
 std::vector<SolverEnum> available_solver_enums() { return solver_enums; }
 
-// collect all the available non-logging solvers
-std::vector<SolverEnum> available_no_logging_solver_enums()
-{
-  std::vector<SolverEnum> enums;
-  for (auto se : solver_enums)
-  {
-    const std::unordered_set<SolverAttribute> & se_attrs =
-        get_solver_attributes(se);
-
-    if (se_attrs.find(LOGGING) == se_attrs.end())
-    {
-      enums.push_back(se);
-    }
-  }
-  return enums;
-}
-
-// collect all the available logging solvers
-std::vector<SolverEnum> available_logging_solver_enums()
+std::vector<SolverConfiguration> available_solver_configurations()
 {
-  std::vector<SolverEnum> enums;
-  for (auto se : solver_enums)
+  std::vector<SolverConfiguration> configs;
+  std::vector<SolverEnum> enums = available_solver_enums();
+  for (SolverEnum e : enums)
   {
-    const std::unordered_set<SolverAttribute> & se_attrs =
-        get_solver_attributes(se);
-
-    if (se_attrs.find(LOGGING) != se_attrs.end())
-    {
-      enums.push_back(se);
-    }
+    SolverConfiguration sc;
+    sc.solver_enum = e;
```

Comment:
I don't feel too strongly about this, but it does seem a bit strange to assign to the member variables this way and implicitly rely on the copy constructor when adding to the vector (so we're not just putting the same object in `configs` twice). An alternative is to add a constructor to the struct that lets you create the object with initialized values: `SolverConfiguration(enum, logging)`. Might be a bit cleaner, even though it's not really different semantically. What do you think?